Added check for maximum allowed WAL file age, configured through last_wal_maximum_age in /etc/barman.conf

georgehansper opened the pull request:
The command 'barman check ...' allows the age of the most recent backup to be checked using `last_backup_maximum_age` in /etc/barman.conf

This change adds a similar option `last_wal_maximum_age` for checking the age of the most recent WAL file archived for the most recent backup.

sammcj commented:
Any update on this?

mnencia commented:
Hi, sorry for the delay replying to this pull request.

Could you please rebase it on current master?

georgehansper commented:
Hi Marco,

Yes, I can update the pull request.
Before I do, I found a bug in the way it works, which I need to investigate further.
I will submit an updated pull request once I have resolved the problem and rebased it to the current version.

Regards,
George Hansper

georgehansper commented:
Hi Marco,

I've updated this fork to the latest master (currently 1.6.0) and fixed the issue I referred to above (where the backup would not start if the 'barman check ...' failed due to the wal or backup being older than the alerting threshold).

georgehansper commented:
Hi Marco,

I've fixed up all the issues - can these changes be merged in ?

Regards,
George

mnencia commented:
Hi George,
thanks for your submission. I'll review it in the next days.

georgehansper commented:
Thanks, Marco

Needless to say, I'm happy to answer any questions about what the changes do and why.
